test/parallel/test-runner-module-mocking.js

```diff
@@ -365,6 +365,24 @@ test('ESM mocking with namedExports option', async (t) => {
   });
 });
 
+test('JSON mocking', async (t) => {
+  await t.test('with defaultExport', async (t) => {
+    const fixturePath = fixtures.path('module-mocking', 'basic.json');
+    const fixture = pathToFileURL(fixturePath);
+    const { default: original } = await import(fixture, { with: { type: 'json' } });
+
+    assert.deepStrictEqual(original, { foo: 'bar' });
+
+    const defaultExport = { qux: 'zed' };
+
+    t.mock.module(fixture, { defaultExport });
+
+    const { default: mocked } = await import(fixture, { with: { type: 'json' } });
```

Comment:
Should we add a check that removing the import attribute would result in a rejected promise?

Reply:
Isn't something deeper in node (or perhaps V8?) is responsible for that?